src/dune_pkg/solver_env.ml

```diff
@@ -54,6 +54,18 @@ module Variable = struct
         of_ordered_set ordered_set
       ;;
 
+      let encode t =
+        Dune_sexp.List
+          (to_list t
+           |> List.map ~f:(fun flag ->
+             (* Note that flags are quoted to make the output a valid Ordered
+                Set Language expression so that [encode] and [decode] can round
+                trip. Technically only the first flag needs to be quoted to
+                form a valid OSL expression but all are quoted for
+                consistency.*)
+             Dune_sexp.Quoted_string (to_string flag)))
```

Comment:
Should be `atom_or_quoted_string` to avoid the quotes.

Reply:
The quotes are necessary because the first element of an OSL expressions must be quoted if it's a string literal (see the comment above this line).